primedev/scripts/client/scriptmodmenu.cpp

```diff
@@ -1,106 +1,76 @@
-		g_pSquirrel<context>->pushstring(sqvm, mod.Name.c_str());
-		g_pSquirrel<context>->arrayappend(sqvm, -2);
-	}
+		g_pSquirrel<context>->pushnewstructinstance(sqvm, 9);
 
-	return SQRESULT_NOTNULL;
-}
+		// name
+		g_pSquirrel<context>->pushstring(sqvm, mod.Name.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 0);
 
-ADD_SQFUNC("bool", NSIsModEnabled, "string modName", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
+		// description
+		g_pSquirrel<context>->pushstring(sqvm, mod.Description.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 1);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
-	{
-		if (!mod.Name.compare(modName))
-		{
-			g_pSquirrel<context>->pushbool(sqvm, mod.m_bEnabled);
-			return SQRESULT_NOTNULL;
-		}
-	}
+		// version
+		g_pSquirrel<context>->pushstring(sqvm, mod.Version.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 2);
 
-	return SQRESULT_NULL;
-}
+		// download link
+		g_pSquirrel<context>->pushstring(sqvm, mod.DownloadLink.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 3);
 
-ADD_SQFUNC("void", NSSetModEnabled, "string modName, bool enabled", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
-	const SQBool enabled = g_pSquirrel<context>->getbool(sqvm, 2);
+		// load priority
+		g_pSquirrel<context>->pushinteger(sqvm, mod.LoadPriority);
+		g_pSquirrel<context>->sealstructslot(sqvm, 4);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
-	{
-		if (!mod.Name.compare(modName))
-		{
-			mod.m_bEnabled = enabled;
-			return SQRESULT_NULL;
-		}
-	}
+		// enabled
+		g_pSquirrel<context>->pushbool(sqvm, mod.m_bEnabled);
+		g_pSquirrel<context>->sealstructslot(sqvm, 5);
 
-	return SQRESULT_NULL;
-}
+		// required on client
+		g_pSquirrel<context>->pushbool(sqvm, mod.RequiredOnClient);
+		g_pSquirrel<context>->sealstructslot(sqvm, 6);
 
-ADD_SQFUNC("bool", NSIsModRemote, "string modName", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
+		// is remote
+		g_pSquirrel<context>->pushbool(sqvm, mod.m_bIsRemote);
+		g_pSquirrel<context>->sealstructslot(sqvm, 7);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
-	{
-		if (!mod.Name.compare(modName))
+		// convars
+		g_pSquirrel<context>->newarray(sqvm);
+		for (ModConVar* cvar : mod.ConVars)
 		{
-			g_pSquirrel<context>->pushbool(sqvm, mod.m_bIsRemote);
-			return SQRESULT_NOTNULL;
+			g_pSquirrel<context>->pushstring(sqvm, cvar->Name.c_str());
+			g_pSquirrel<context>->arrayappend(sqvm, -2);
 		}
-	}
-
-	return SQRESULT_NULL;
-}
+		g_pSquirrel<context>->sealstructslot(sqvm, 8);
```

Comment:
Maybe make all of this its own function and make it possible to get the modinfo for a singular mod as well?

Reply:
Do you mean adding a `array<ModInfo> getModInformation( string modName )` method here? That could do it.

Reply:
Yea but it dosnt need to return an array

Reply:
User might have multiple versions of a same mod, hence the array, don't you think?

Reply:
Yes i thought we could just add an array for the version field but thinking about it what if other fields are different between versions as well